packages/taco/src/config-validator.ts

```diff
@@ -0,0 +1,279 @@
+   * - Provider/signer presence validation
+   * - Chain compatibility check (if chain info is available synchronously)
+   *
+   * @param {TacoClientConfig} config - Configuration to validate
+   * @returns {ValidationResult} Validation result with isValid boolean and errors array
+   */
+  static validateFast(config: TacoClientConfig): ValidationResult {
+    const errors: string[] = [];
+
+    // Validate domain
+    if (!config.domain) {
+      errors.push('The property `domain` is required');
+    } else if (!this.isValidDomain(config.domain)) {
+      errors.push(
+        `Invalid domain name: ${config.domain}. Supported domains: ${this.getSupportedDomains().join(', ')}`,
+      );
+    }
+
+    // Validate ritual ID
+    if (!config.ritualId) {
+      errors.push('The property `ritualId` is required');
+    } else if (!this.isValidRitualId(config.ritualId)) {
+      errors.push(
+        `Invalid ritual ID: ${config.ritualId} for domain ${config.domain}`,
+      );
+    }
+
+    // Validate blockchain client configuration
+    if ('viemClient' in config) {
+      // Viem configuration
+      if (!config.viemClient) {
+        errors.push('viemClient is required for viem configuration');
+      }
+      if (!config.viemSignerAccount) {
+        errors.push('viemSignerAccount is required for viem configuration');
+      }
+    } else if ('ethersProvider' in config) {
+      // Ethers configuration
+      if (!config.ethersProvider) {
+        errors.push('ethersProvider is required for ethers configuration');
+      }
+      if (!config.ethersSigner) {
+        errors.push('ethersSigner is required for ethers configuration');
+      }
+    } else {
+      errors.push(
+        'Configuration must include either viem objects (viemClient + viemSignerAccount) or ethers objects (ethersProvider + ethersSigner)',
+      );
+    }
+
+    // Validate chain compatibility (synchronous check)
+    const chainValidation = this.validateChainCompatibility(config);
+    if (!chainValidation.isValid) {
+      errors.push(...chainValidation.errors);
+    }
+
+    return { isValid: errors.length === 0, errors };
+  }
+
+  /**
+   * Synchronous chain compatibility validation
+   *
+   * Validates provider chain compatibility with domain requirements using
+   * synchronously available chain information.
+   *
+   * @param {TacoClientConfig} config - Configuration to validate
+   * @returns {ValidationResult} Validation result
+   * @private
+   */
+  private static validateChainCompatibility(
+    config: TacoClientConfig,
+  ): ValidationResult {
+    const errors: string[] = [];
+
+    // Get expected chain ID for domain
+    const expectedChainId = this.getExpectedChainId(config.domain);
+    if (!expectedChainId) {
+      errors.push(`Unsupported domain: ${config.domain}`);
+      return { isValid: false, errors };
+    }
+
+    // Check viem client chain compatibility
+    if ('viemClient' in config && config.viemClient) {
```

Comment:
Can this be `isViemClient`?

If i'm grasping this properly, It feels like you need a base common validate config helper that your various validation/check functions that take a config call first, and then do their own validation subsequently. That way you don't have to repeatedly check existence of various properties.